packages/dynamite/dynamite_runtime/lib/src/utils/array_checker.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:built_collection/built_collection.dart';
+
+/// Checks the [input] length against [minItems].
+///
+/// Throws an `Exception` containing the [parameterName] if the `input` is too short.
+void checkMinItems(BuiltList<dynamic>? input, int minItems, String parameterName) {
```

Comment:
Why not pass the `ListBuilder`?
This would save us a build and thus a list copy that would get discarded immediately.

The builder also has a length property: https://pub.dev/documentation/built_collection/latest/built_collection/ListBuilder/length.html

Reply:
The problem is that for parameters we get a BuiltList and for responses we get a ListBuilder. One of them has to be converted to the other, or we handle it special and make the checks different such that lists only pass the length property like you said (but that requires more work).

Reply:
I guess it would be possible to do this without a lot of hacks by just calling `$value.length` which will work for both options.

Reply:
1. do we make use of this anywhere? I can't find any reference so please add a test to e2e.
2. What about making the parameter of the `checkMinItems` and `checkMaxItems` methods dynamic and just do a typecheck? This shouldn't be as bad as it sounds because the methods are only used by generated code.

Reply:
1. I added e2e tests already :thinking:
2. I think passing the length to the method directly is better than doing a type check inside the method. We know that only BuiltList and ListBuilder will be passed, so we can safely call `.length` on them.

Reply:
I meant a test where a `BuiltList` is used.
I can only see e2e tests where a `ListBuilder` is used.

> It should be possible to do this without a lot of hacks by just calling $value.length which will work for both options (somehow I thought this would be more complex).

Yeah this might be the best thing for now. I just wanted to not have the extra complexity  in dynamite but there is probably nothing we can do for now :(